lib/internal/streams/destroy.js

```diff
@@ -124,12 +150,58 @@ function errorOrDestroy(stream, err) {
 
   if ((r && r.autoDestroy) || (w && w.autoDestroy))
     stream.destroy(err);
-  else if (needError(stream, err))
-    stream.emit('error', err);
+  else if (needError(stream, err)) {
+    if (sync) {
+      process.nextTick(emitErrorNT, err);
+    } else {
+      stream.emit('error', err);
+    }
+  }
 }
 
+function construct(stream, options, cb) {
+  if (typeof stream._construct !== 'function') {
+    return;
+  }
+
+  const r = stream._readableState;
+  const w = stream._writableState;
+  const s = w || r;
```

Comment:
Is there a reason why you are assigning this to `w`  instead of `r`? If so, can you document it?

Reply:
No, just had to pick one of them. It's unnecessary to read/write the state of duplex streams on both sides.